client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -1,11 +1,21 @@
+import { useState, useEffect } from 'react';
 import { Redirect, Route, RouteProps } from 'react-router';
 import { useAuth } from '../../context/useAuthContext';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { useHistory } from 'react-router-dom';
 
 export default function ProtectedRoute(routeProps: RouteProps) {
   const { loggedInUser } = useAuth();
-  if (loggedInUser) {
+  const [authenticated, setAuth] = useState<boolean>(false);
```

Comment:
```suggestion
  const [authenticated, setAuthenticated] = useState<boolean>(false);
```
What is the reason for having a state here? The previous implementation of the protected route using the logged in user object (from AuthContext) accomplished the same in a simpler way :)